src/structure/pdb.jl

```diff
@@ -30,26 +39,231 @@ end
+if the keyword argument `overwrite` is set `true`, then it will overwrite the PDB file if exists in the `pdb_dir`
+"""
+function downloadentirepdb(;pdb_dir::AbstractString=pwd(), overwrite::Bool=false)
+    # Get the list of all pdb entries from RCSB PDB Server using getallpdbentries() and downloads them
+    downloadmultiplepdb(getallpdbentries(), pdb_dir=pdb_dir, overwrite=overwrite)
+end
+
+
+"""
+Updates your local copy of the PDB files. It gets the weekly lists of new, 
+modified and obsolete pdb entries and automatically downloads those PDB files to the `pdb_dir`;
+if the keyword argument `pdb_dir` is set the PDB files in the specified directory are updated;
+"""
+function updatepdb(;pdb_dir::AbstractString=pwd())
+    addedlist, modifiedlist, obsoletelist = getrecentchanges()
+    # download the newly added and modified pdb files 
+    downloadmultiplepdb(vcat(addedlist,modifiedlist), pdb_dir=pdb_dir, overwrite=true)
+    # set the obsolete directory to be inside pdb_dir
+    obsolete_dir=joinpath(pdb_dir,"obsolete")
+    for pdbid in obsoletelist
+        oldfile = joinpath(pdb_dir,"$pdbid.pdb")
+        newfile = joinpath(obsolete_dir, "$pdbid.pdb")
+        # if obsolete pdb is in the "pdb_dir", move it to "obsolete" directory inside "pdb_dir" 
+        if isfile(oldfile)
+            if !isdir(obsolete_dir)
+                mkpath(obsolete_dir)
+            end
+            mv(oldfile,newfile)
+        # if obsolete pdb is already in the obsolete directory, inform the user and skip
+        elseif isfile(newfile)
+            info("PDB $pdbid is already moved to the obsolete directory")
+        # if obsolete pdb not available in both pdb_dir and obsolete, inform the user and skip
+        else
+            info("Obsolete PDB $pdbid is missing")
+        end
     end
 end
 
 
+"""
+Download all obsolete PDB files from RCSB PDB server not present in the local obsolete PDB directory;
+if the keyword argument `obsolete_dir` is set the obsolete PDB files are downloaded to the specified directory;
+"""
+function downloadallobsoletepdb(;obsolete_dir::AbstractString=pwd())
+    # Get all obsolete PDB files in RCSB PDB Server using getallobsolete() and download them
+    obsoletelist = getallobsolete()
+    downloadmultiplepdb(obsoletelist, pdb_dir=obsolete_dir)
+end
+
+
+"""
+Download a PDB file or biological assembly from the RCSB PDB server and reads it. 
+By default downloads the PDB file to current working directory; 
+if the keyword argument `ba_number` is set the biological assembly with that number will be downloaded; 
+if the keyword argument `pdb_dir` is set the PDB file is downloaded to the specified directory; 
+if the keyword argument `obsolete` is set `true`, the PDB file is downloaded into the obsolete directory inside `pdb_dir`;
+if the keyword argument `overwrite` is set `true`, then it will overwrite the PDB file if it exists in the `pdb_dir`;
+"""
+function retrievepdb(pdbid::AbstractString;
```

Comment:
Ya. I was so confused. Then finally decided keep it as in BioPython. So if BioPython users are using this, they may find it easy. As mentioned `downloadpdbandread` will be too long. We will keep it as `retrievepdb` as of now. Any other good function name is welcomed.